metric/registry/registry.go

```diff
@@ -112,7 +138,7 @@ func Compatible(candidate, existing metric.Descriptor) bool {
 // registration, this returns the already-registered instrument.  If
 // there is no conflict and no prior registration, returns (nil, nil).
 func (u *uniqueInstrumentMeterImpl) checkUniqueness(descriptor metric.Descriptor) (metric.InstrumentImpl, error) {
-	impl, ok := u.state[keyOf(descriptor)]
+	impl, ok := u.state[descriptor.Name()]
```

Comment:
Why are we able to lessen the restriction here and only check if the name is not already registered instead of the full descriptor?

Reply:
The former code had a 1-level map by (instrument name, library name, version). 

There are now two levels of mapping, the `MeterProvider` maintains a unique mapping of a 3-part key (instrumentation name, version, schema_url) to Meter. This level of the mapping ensures all Meters share underlying state.

This object, adapted from the prior code, maintains a 1-part key which is just instrument name. At this level in the code, the mapping ensures all instruments are registered with the same type.